Rules/UseCorrectCasing.cs

```diff
@@ -52,6 +53,12 @@ public override IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string file
                 var fullyqualifiedName = $"{commandInfo.ModuleName}\\{shortName}";
                 var isFullyQualified = commandName.Equals(fullyqualifiedName, StringComparison.OrdinalIgnoreCase);
                 var correctlyCasedCommandName = isFullyQualified ? fullyqualifiedName : shortName;
+                // For binaries that could exist on both Windows and Linux like e.g. git we do not want to expand
+                // git to git.exe to keep the script cross-platform compliant
+                if (commandInfo.CommandType == CommandTypes.Application && correctlyCasedCommandName.EndsWith(".exe"))
```

Comment:
What about `.cmd`, `.bat`, `.com`? Things like `code.cmd` or `more.com` would benefit from this. Not sure about other `%PATHEXT%` extensions...

Reply:
Also:

```csharp
correctlyCasedCommandName = System.IO.Path.GetFilenameWithoutExtension(correctlyCasedCommandName);
```

Reply:
Good point. I was afraid at first that removing the file extension generically would cause problems in special cases where e.g. the executable contains periods (e.g. the `Microsoft.Uev.CscUnpinTool` binary). But it turns out this is not a problem because the short name of the CommandInfo of `Microsoft.Uev.CscUnpinTool` is `Microsoft.Uev.CscUnpinTool.exe`, therefore we always have an extension to remove on Windows.
There are a few edge cases on Linux though (e.g. `Invoke-Formatter telnet.netkit`) and this would totally break on WSL, therefore we have make this code Windows-only I think

Reply:
Yeah I think Windows-only is the way to go